src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -4033,6 +4033,10 @@ protected JCClassDecl enumDeclaration(JCModifiers mods, Comment dc) {
 
         Name name = typeName();
 
+        if(typeParametersOpt().size() > 0) {
```

Comment:
Nit: we usually have a space between `if` and `(`.
Nit: using `!.isEmpty()` might be better here (it would also be slightly faster for non-empty Lists, but since that would only happen in erroneous cases, it is not so critical).